### PR TITLE
bpo-1635741 port cmath to multi-phase init (pep 489)

### PR DESCRIPTION
**Pull request by koubaa**
https://bugs.python.org/issue1635741

### CODE REVIEW

**Comment by koubaa** [comment]
@vstinner @shihai1991 @corona10 please review

File: Modules/cmathmodule.c [hunk 1]
```diff
@@ -1254,37 +1254,34 @@ static PyMethodDef cmath_methods[] = {
...
-        return NULL;
-
-    PyModule_AddObject(m, "pi",
-                       PyFloat_FromDouble(Py_MATH_PI));
-    PyModule_AddObject(m, "e", PyFloat_FromDouble(Py_MATH_E));
-    PyModule_AddObject(m, "tau", PyFloat_FromDouble(Py_MATH_TAU)); /* 2pi */
-    PyModule_AddObject(m, "inf", PyFloat_FromDouble(m_inf()));
-    PyModule_AddObject(m, "infj", PyComplex_FromCComplex(c_infj()));
+    if (PyModule_AddObject(m, "pi",
+                       PyFloat_FromDouble(Py_MATH_PI)) < 0) {
```
Comment (vstinner) [inline Modules/cmathmodule.c]:
```suggestion
    if (PyModule_AddObject(m, "pi", PyFloat_FromDouble(Py_MATH_PI)) < 0) {
```

File: Modules/cmathmodule.c [hunk 2]
```diff
@@ -1254,37 +1254,34 @@ static PyMethodDef cmath_methods[] = {
...
-    NULL,
-    NULL,
-    NULL,
-    NULL
-};
-
-PyMODINIT_FUNC
-PyInit_cmath(void)
+static int
+cmath_exec(PyObject *m)
```
Comment (vstinner) [inline Modules/cmathmodule.c]: Since you modify all lines using "m", can you please rename it to "mod"?

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**